### PR TITLE
Feat: final feeds page

### LINKED ISSUE #494: Final Feeds search page
### Describe the problem

[Final Feeds page design](https://www.figma.com/design/LZVQecGplb4Lnuh7a24Jef/GTFS-Tool-Forms?node-id=2457-9866&t=sh0WikiRbjcUP4jv-1)

### Proposed solution

Update the search page, including 

- [ ]  Updating the search result components to be a table
- [ ] Add search button so search is done after search button is clicked, rather than autosearch. When the user presses enter, they should complete their search query and run a search.
- [ ]  When there are multiple transit providers within 1 result, only display the first one before the comma separation. (There may be cases where this prematurely shortens a transit provider result — that is a fine tradeoff here.) 
- [ ] When the status of a feed is deprecated, display it in the transit provider column as shown in the design composite. 
- [ ] Add pagination
- ~~Cypress tests, mocking the endpoints~~

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Alessandro100**
Closes #494 and #500
**Summary:**

Implements the design for the feeds page + adds loading state + error state + empty state + pagination

**Expected behavior:** 

When searching for feeds, it's expected that the elements show up in the table
- Search query should appear in url
- Pagination number should appear in url
- Should be able to toggle gtfs schedule / realtime
- When no results -> show empty state
- Default state is a search for the entire dataset (20 elements at a time)

**Testing tips:**

Go to the feeds page from the hosted url and play around with search and all it's settings (search, pagination, navigation)

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

Screenshot 2024-06-27 at 08 06 25
Screenshot 2024-06-27 at 08 06 08
Screenshot 2024-06-27 at 08 06 00
Screenshot 2024-06-27 at 08 05 48
Screenshot 2024-06-28 at 09 17 00

### CODE REVIEW

**Comment by emmambd** [comment]
- Noticing that our default page for search has a lot of deprecated feeds: https://mobility-feeds-dev--pr-514-k371l82f.web.app/feeds. Is it possible to filter out the feeds in the default view with a status of deprecated, so it's a more intuitive experience for users initially? It doesn't look like these are in alphabetical order by provider atm, so curious what the default sort is.
- Looks like the full list of transit providers is displayed: 

Screenshot 2024-07-02 at 9 09 31 AM

Wondering if there was discussion about this line from the original #494 : 

> 
When there are multiple transit providers within 1 result, only display the first one before the comma separation. (There may be cases where this prematurely shortens a transit provider result — that is a fine tradeoff here.)
- Can we left justify instead of right when the content doesn't fit on 1 line?
Screenshot 2024-07-02 at 9 12 07 AM

- Lastly, I'm wondering if we should modify the spacing of the Data Type section, maybe so the filters are in alignment with the column names? The original design looks slightly off to me now. @Alessandro100 curious for your thoughts

**Comment by qcdyx** [comment]
Hey @Alessandro100, the bug I mentioned in Slack still persists. 

1. Goto https://mobility-feeds-dev--pr-514-k371l82f.web.app/feeds
2. Click Login and Sign in with Google 
4. Click Feeds in header
5. search for california and hit Search button -> No results show up
image
Not sure if it's related to the lack of a global user state management reported in #516

**Comment by Alessandro100** [comment]
> Hey @Alessandro100, the bug I mentioned in Slack still persists.
> 
> 1. Goto https://mobility-feeds-dev--pr-514-k371l82f.web.app/feeds
> 2. Click Login and Sign in with Google
> 3. Click Feeds in header
> 4. search for california and hit Search button -> No results show up
>    image
>    Not sure if it's related to the lack of a global user state management reported in [Lack a global mechanism for managing user states during cold loading #516](https://github.com/MobilityData/mobility-feed-api/issues/516)

Great find, I tested it locally and when you sign into google it seems like there is an error and the user.token becomes undefined. This seems to be a bug not caused by this feature, and should be investigated in a separate ticket

**Comment by Alessandro100** [comment]
> * Noticing that our default page for search has a lot of deprecated feeds: https://mobility-feeds-dev--pr-514-k371l82f.web.app/feeds. Is it possible to filter out the feeds in the default view with a status of deprecated, so it's a more intuitive experience for users initially? It doesn't look like these are in alphabetical order by provider atm, so curious what the default sort is.
> * Looks like the full list of transit providers is displayed:
> 
> Screenshot 2024-07-02 at 9 09 31 AM
> Wondering if there was discussion about this line from the original #494 :
> 
> > 
> 
> When there are multiple transit providers within 1 result, only display the first one before the comma separation. (There may be cases where this prematurely shortens a transit provider result — that is a fine tradeoff here.)
> 
> * Can we left justify instead of right when the content doesn't fit on 1 line?
> 
> Screenshot 2024-07-02 at 9 12 07 AM
> * Lastly, I'm wondering if we should modify the spacing of the Data Type section, maybe so the filters are in alignment with the column names? The original design looks slightly off to me now. @Alessandro100 curious for your thoughts

1. The initial search and deprecation values would have to come from additions in the search endpoint (coming in future tickets) 
2.  Improved provider names done
3. Feed Description change done
4. Data Type styling is ok in my opinion

**Comment by emmambd** [comment]
@Alessandro100 For #1 - I'm confused how this is a search endpoint improvement. Isn't it a different filter setting in the front end?

**Comment by emmambd** [comment]
@qcdyx Could you make an issue for the bug?

**Comment by Alessandro100** [comment]
> @Alessandro100 For #1 - I'm confused how this is a search endpoint improvement. Isn't it a different filter setting in the front end?

we don't load all the feeds and let the frontend sort it out. The backend will only return 20 feeds out of ~2k, so even if I sort in the frontend it will only sort the local 20. Similar issue with the deprecated feed, I can filter them out but only based on the 20 I get from the backend.

If we want to get 20 non deprecated feeds, that logic would have to be done in the backend same with the sort results

**Comment by qcdyx** [comment]
> @qcdyx Could you make an issue for the bug?

Yes,  resolving #516 fix this bug and #443.

File: web-app/src/app/screens/Feeds/index.tsx [hunk 1]
```diff
@@ -186,90 +204,142 @@ export default function Feed(): React.ReactElement {
...
-                      </Button>
+                      <h3>{t('common:errors.generic')}</h3>
+                      <Typography>
+                        {t('common:errors.checkInternet')},{' '}
+                        <a href='mailto:api@mobilitydata.org'>
+                          {t('common:contactUs')}
+                        </a>{' '}
+                        for
+                        {t('common:errors.forAssistance')}
+                      </Typography>
```
[the remark this answers is not in the document]
Comment (Alessandro100) [reply web-app/src/app/screens/Feeds/index.tsx]: great comment, I discovered the translation library has the `Trans` component to help with situations like this

**Review by davidgamez — APPROVED** [review]
LGTM!

**Pull request merged**